client/js/widgets/label.js

```diff
@@ -1,5 +1,37 @@
 import { Widget } from './widget.js';
 
+export async function unsetHTML(w) {
+  document.getElementById(w.id).innerHTML = null;
+	w.input = document.createElement('textarea');
+	 
+	w.addDefaults({
+    height: 20,
+    movable: false,
+    layer: -2,
+    typeClasses: 'widget label',
+
+    text: '',
+    editable: false,
+    twoRowBottomAlign: false
+    });
+  w.domElement.appendChild(w.input);
+	w.input.value = w.get('text'); 
+  w.input.addEventListener('keyup', e=>w.setText(e.target.value));  
+  }
+ 
+  export async function setHTML(w) {
+    if(w.get('html') == true) {
+      var HTMLtext = w.get('text')
+	    var HTMLtext = HTMLtext.toString()
+	    .replace(/\*\*(.*?)\*\*/gim, '<b>$1</b>')
+	    .replace(/\*(.*?)\*/gim, '<i>$1</i>')
+      .replace(/\-\-(.*?)\-\-/gim, '<sub>$1</sub>')
+   	  .replace(/\^\^(.*?)\^\^/gim, '<sup>$1</sup>')
+      .replace(/<[^A-Za-z0-9]*a +href(.*?)>/im, '<a href=https://github.com/ArnoldSmith86/virtualtabletop/wiki/Widget-Properties#label>')
+	    .replace(/<[^A-Za-z0-9]*iframe(.*?)>/im, '<iframe src=https://virtualtabletop.io/Tutorials> height="500" width="500"')
```

Comment:
Using markdown syntax is fine but I wouldn't allow just any HTML. You didn't disallow `<script>` at all for example and there are probably 20 ways to get around your blacklist.

Reply:
If people even like this idea, then what is and isn't allowed will need to be a discussion.  I just wanted to throw in a couple of things as an example really.  Temporarily I updated this with some replaces that will get rid of <script> tags and any tag with parenthesis inside the tag.  I think the way I am doing markdown could be used to white-list specific tags if that was the way people wanted to go, and then any tag could be stripped from the text.  I'm probably not going to mess with this part much more unless people want this feature and a consensus can be reached about what to include.